source/opt/reduce_load_size.cpp

```diff
@@ -139,6 +139,8 @@ bool ReduceLoadSize::ShouldReplaceExtract(Instruction* inst) {
 
   all_elements_used =
       !def_use_mgr->WhileEachUser(op_inst, [&elements_used](Instruction* use) {
+        if (use->GetOpenCL100DebugOpcode() != OpenCLDebugInfo100InstructionsMax)
+          return true;
```

Comment:
I think this predicate will be used quite a bit.  During traversal, we may need to skip debug instructions or ignore them for reasons similar to this.  Perhaps factor this into a IsDebugInstr() predicate?

Reply:
Do you mean adding some code similar to the following function in a common header like `source/opt/reflect.h`
```
bool IsDebugInstr(Instruction *instr) {
  return instr->GetOpenCL100DebugOpcode() != OpenCLDebugInfo100InstructionsMax;
}
```
?

Reply:
Yes, though not as a free function.  As a member of `class Instruction`.

Reply:
done